head.go

```diff
@@ -195,8 +227,30 @@ func (h *Head) ReadWAL() error {
 
 	// Track number of samples that referenced a series we don't know about
 	// for error reporting.
-	var unknownRefs int
+	var unknownRefs uint64
+
+	// Start workers that each process samples for a partition of the series ID space.
+	// They are connected through a ring of channels which ensures that all sample batches
+	// read from the WAL are processed in order.
+	var (
+		n          = runtime.GOMAXPROCS(0)
```

Comment:
Can we use `NumCPU`? https://golang.org/pkg/runtime/#GOMAXPROCS

Reply:
I think this was done intentionally to use the configured go procs.

Reply:
By default `GOMAXPROCS` will be set to `NumCPU()`. However, if it is set manually to a different number, we should respect that. The runtime will only schedule onto `GOMAXPROCS` threads at a time so anything above that only introduces more active goroutines requiring coordination overhead even though they cannot actually run in parallel.